kernel/data-pipeline/scenario/cdc/client/src/main/java/org/apache/shardingsphere/data/pipeline/cdc/client/CDCClient.java

```diff
@@ -93,14 +82,174 @@ protected void initChannel(final NioSocketChannel channel) {
+            throw new GetResultTimeoutException("Get result timeout");
+        }
+        if (!Strings.isNullOrEmpty(responseFuture.getErrorMessage())) {
+            throw new IllegalArgumentException(String.format("Get Response failed, reason: %s", responseFuture.getErrorMessage()));
+        }
+        return responseFuture.getResult();
+    }
+    
+    /**
+     * Start streaming.
+     *
+     * @param parameter parameter
+     * @return streaming id
+     * @throws IllegalStateException     start streaming failed
+     * @throws GetResultTimeoutException get result timeout
+     */
+    public String startStreaming(final StartStreamingParameter parameter) {
+        StreamDataRequestBody streamDataRequestBody = StreamDataRequestBody.newBuilder().setDatabase(parameter.getDatabase()).setFull(parameter.isFull())
+                .addAllSourceSchemaTable(parameter.getSchemaTables()).build();
+        String requestId = RequestIdUtils.generateRequestId();
+        CDCRequest request = CDCRequest.newBuilder().setRequestId(requestId).setType(Type.STREAM_DATA).setStreamDataRequestBody(streamDataRequestBody).build();
+        ClientConnectionContext connectionContext = channel.attr(ClientConnectionContext.CONTEXT_KEY).get();
+        ResponseFuture responseFuture = new ResponseFuture();
+        connectionContext.getResponseFutureMap().put(requestId, responseFuture);
+        channel.writeAndFlush(request);
+        try {
+            String streamingId = getResultFromResponseFuture(responseFuture).toString();
+            log.info("Start streaming success, streaming id: {}", streamingId);
+            return streamingId;
+        } finally {
+            connectionContext.getResponseFutureMap().remove(requestId);
+        }
+    }
+    
+    /**
+     * Restart streaming.
+     *
+     * @param streamingId streaming id
+     * @throws IllegalStateException the channel is not active
+     */
+    public void restartStreaming(final String streamingId) {
+        if (checkStreamingIdExist(streamingId)) {
+            stopStreaming(streamingId);
+        }
+        String requestId = RequestIdUtils.generateRequestId();
+        StartStreamingRequestBody body = StartStreamingRequestBody.newBuilder().setStreamingId(streamingId).build();
+        CDCRequest request = CDCRequest.newBuilder().setRequestId(requestId).setType(Type.START_STREAMING).setStartStreamingRequestBody(body).build();
+        ResponseFuture responseFuture = new ResponseFuture();
+        ClientConnectionContext connectionContext = channel.attr(ClientConnectionContext.CONTEXT_KEY).get();
+        connectionContext.getResponseFutureMap().put(requestId, responseFuture);
+        channel.writeAndFlush(request);
+        try {
+            getResultFromResponseFuture(responseFuture);
+            log.info("Restart streaming success, streaming id: {}", streamingId);
+        } finally {
+            connectionContext.getResponseFutureMap().remove(requestId);
+        }
```

Comment:
Could we merge finally block code into `getResultFromResponseFuture`? Make sure it could be done automatically.